source/server/listener_impl.h

```diff
@@ -74,12 +76,128 @@ class ListenSocketFactoryImpl : public Network::ListenSocketFactory,
 // TODO(mattklein123): Consider getting rid of pre-worker start and post-worker start code by
 //                     initializing all listeners after workers are started.
 
+/**
+ * The immutable factory context could be used by listener filter factory contexts and network
+ * filter factory contexts.
+ */
+class ListenerFactoryContextBaseImpl final : public Configuration::FactoryContext,
```

Comment:
This is per listener right? Let's call this `PerListenerFactoryContextImpl`. 

The one below is per filter chain? Let's call that `PerFilterChainFactoryContextImpl`.

Reply:
This `ListenerFactoryContextBaseImpl` is the common part of below per listener and per filter chain context. It wraps the `Server` instance and a `ListenerImpl` instance.

The below `ListenerFactoryContextImpl` is per listener, I renamed to `PerListenerFactoryContextImpl`
The `FilterChainFactoryContextImpl` is per filter chain and I renamed to `PerFilterChainFactoryContextImpl`